xrootd/client/filesystem_test.go

```diff
@@ -110,3 +110,65 @@ func TestFileSystem_Open(t *testing.T) {
 		}
 	}
 }
+
+func testFileSystem_Truncate(t *testing.T, addr string) {
+	filePath := "/tmp/test_truncate_fs.txt"
+	write := []uint8{1, 2, 3, 4, 5, 6, 7, 8}
+	want := write[:4]
+
+	client, err := NewClient(context.Background(), addr, "gopher")
+	if err != nil {
+		t.Fatalf("could not create client: %v", err)
+	}
+	defer client.Close()
+
+	fs := client.FS()
+	file, err := fs.Open(context.Background(), filePath, xrdfs.OpenModeOwnerWrite, xrdfs.OpenOptionsOpenUpdate)
+	if err != nil {
+		t.Fatalf("invalid open call: %v", err)
+	}
+	defer file.Close(context.Background())
+
+	_, err = file.WriteAt(write, 0)
+	if err != nil {
+		t.Fatalf("invalid write call: %v", err)
+	}
+
+	err = file.Sync(context.Background())
+	if err != nil {
+		t.Fatalf("invalid sync call: %v", err)
+	}
+
+	file.Close(context.Background())
+
+	err = fs.Truncate(context.Background(), filePath, int64(len(want)))
+	if err != nil {
+		t.Fatalf("invalid truncate call: %v", err)
+	}
+
+	file, err = fs.Open(context.Background(), filePath, xrdfs.OpenModeOwnerRead, xrdfs.OpenOptionsOpenRead)
+	defer file.Close(context.Background())
```

Comment:
please check for the returned `err`.
the open shouldn't fail, but... there's a network between :) (and possibly an ocean)